.github/workflows/ci-preview.yaml

```diff
@@ -0,0 +1,342 @@
+name: ci - build previews
+
+on:
+  workflow_call:
+    inputs:
+      pr_number:
+        type: string
+        required: false
+      commit_sha:
+        type: string
+        required: true
+      dashboard_deploy:
+        type: boolean
+        required: false
+        default: false
+      docs_deploy:
+        type: boolean
+        required: false
+        default: false
+      rustdoc_deploy:
+        type: boolean
+        required: false
+        default: false
+    secrets:
+      build_buddy_token:
+        required: true
+      vercel_token:
+        required: false
+      vercel_org_id:
+        required: false
+      vercel_project_id_dashboard:
+        required: false
+      vercel_project_id_docs:
+        required: false
+      vercel_project_id_rustdoc:
+        required: false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  dashboard:
+    if: ${{ inputs.dashboard_deploy }}
+    runs-on: ubicloud-standard-8-ubuntu-2404
+    env:
+      ARTIFACT_NAME: "dashboard"
+      PR_NUMBER: ${{ inputs.pr_number }}
+      DEPLOY_DOCS: ${{ inputs.dashboard_deploy }}
+      COMMIT_SHA: ${{ inputs.commit_sha }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install zstd
+        run: sudo apt-get update && sudo apt-get install zstd -y
+
+      - uses: ./.github/actions/setup-bazel
+        with:
+          build_buddy_token: ${{ secrets.build_buddy_token }}
+
+      - name: Build
+        run: bazel build //cloud/dashboard
+
+      - name: Compress
+        run: |
+          pwd=$(pwd)
+          cd ./target-bazel/bin/cloud/dashboard/build
+          tar -cf - * | zstd -o "$pwd/app-scuffle-cloud.tar.zst"
+
+      - name: Upload
+        if: ${{ env.ARTIFACT_NAME != '' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.ARTIFACT_NAME }}
```

Comment:
Lets just always upload this and call it `ci-preview-rustdoc` `ci-preview-docs` `ci-preview-dashboard` You also will need to update the `release.yaml` file to use the `ci-preview-rustdoc` too